### PR TITLE
Python client packaging

### PR DESCRIPTION
**Pull request by lpascal-ledger**

### CODE REVIEW

File: client/MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1 @@
+include src/ledger_app_clients/ethereum/keychain/*
```
Comment (fbeutin-ledger) [inline client/MANIFEST.in:1]: newline at end of file

File: .github/workflows/python-client.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Python client checks, package build and deployment
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - develop
+      - master
+    paths:
+      - ./client/
```
Comment (fbeutin-ledger) [inline .github/workflows/python-client.yml:10]: `python_client` ?

File: .github/workflows/python-client.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    needs: [lint, mypy]
+    name: Build, test and deploy the Python package
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_pypi_deployment.yml@pypi_deployment
+    with:
+      package_directory: "client/"
+      stable_deployment: ${{ github.ref == 'refs/heads/master' }}
+      check_changelog_version: true
+      publish: ${{ github.event_name == 'push' }}
+    secrets:
+      pypi_token: ${{ github.ref == 'refs/heads/master' && secrets.PYPI_PUBLIC_API_TOKEN  || secrets.TEST_PYPI_PUBLIC_API_TOKEN }}
```
Comment (fbeutin-ledger) [inline .github/workflows/python-client.yml:48]: Not sure I understand this line ? Won't this evaluate a true/false boolean ?

File: client/MANIFEST.in (hunk 1 above)
Comment (fbeutin-ledger) [reply client/MANIFEST.in:1]: Also what does this line mean and what does this file do ?

File: client/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Ethereum app Python client
+
+This package allows to communicate with the Ethereum application, either on a
```
Comment (fbeutin-ledger) [inline client/README.md:3]:
```suggestion
This package allows to communicate with the Ledger Ethereum application, either on a
```

File: client/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+real device, or emulated on Speculos.
```
Comment (fbeutin-ledger) [inline client/README.md:4]: (120 char lines is really ok for a markdown file)

File: client/README.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Ethereum app Python client
```
Comment (fbeutin-ledger) [inline client/README.md:1]:
```suggestion
# Python client for the Ledger Ethereum application 
```

File: client/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+This package is deployed:
+
+- on `pypi.org` for the stable version. This version will work with the
+  application available on the `master` branch.
+  ```bash
+  pip install ledger_app_clients.ethereum
+  ```
+- on `test.pypi.org` for the rolling release. This version will work with the
+  application code on the `develop` branch.
```
Comment (fbeutin-ledger) [inline client/README.md:16]:
```suggestion
- on `test.pypi.org` for the rolling release. This version will work with the
  application available on the `develop` branch.

```

File: client/README.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+This package allows to communicate with the Ethereum application, either on a
+real device, or emulated on Speculos.
+
+## Installation
+
+This package is deployed:
+
+- on `pypi.org` for the stable version. This version will work with the
+  application available on the `master` branch.
```
Comment (fbeutin-ledger) [inline client/README.md:11]:
```suggestion
- on `pypi.org` for the stable version. This version will work with the
  application available on the `master` branch.

```

File: client/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  ```bash
+  pip install --extra-index-url https://test.pypi.org/simple/ ledger_app_clients.ethereum
+  ```
+
+### Installation from sources
+
+You can install the client from this repo:
+
+```bash
+cd client/
```
Comment (fbeutin-ledger) [inline client/README.md:26]: We are already in client

File: client/MANIFEST.in (hunk 1 above)
Comment (lpascal-ledger) [reply client/MANIFEST.in:1]: This is needed to include the text files stored into this path into the Python package, else only `.py` file would be included.
This files are needed by the `keychain,py` module.

File: .github/workflows/python-client.yml (hunk 2 above)
Comment (lpascal-ledger) [reply .github/workflows/python-client.yml:10]: For now, the python client is stored in the `./client` directory,

File: .github/workflows/python-client.yml (hunk 3 above)
Comment (lpascal-ledger) [reply .github/workflows/python-client.yml:48]: It seems it is an accepted 'fake' ternary conditional operator on GitHub actions.
References:
- https://github.com/actions/runner/issues/409#issuecomment-752775072
- https://github.com/orgs/community/discussions/26738#discussioncomment-3836389

Comment (lpascal-ledger) [reply .github/workflows/python-client.yml:48]: (but yeah it needs to be properly deployed to assert it works, as for now current CI never tried to push a package so far)

**Review by fbeutin-ledger — APPROVED** [review]

**Pull request merged**